examples/custom_client_tls.rs

```diff
@@ -0,0 +1,104 @@
+// Custom client supporting both native-tls and rustls-tls
+// Run with `USE_RUSTLS=1` to pick rustls.
+use std::sync::Arc;
+
+use futures::{StreamExt, TryStreamExt};
+use hyper::client::HttpConnector;
+use k8s_openapi::api::core::v1::Pod;
+use serde_json::json;
+use tower::ServiceBuilder;
+
+use kube::{
+    api::{Api, DeleteParams, ListParams, PostParams, ResourceExt, WatchEvent},
+    service::SetBaseUriLayer,
+    Client, Config,
+};
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    std::env::set_var("RUST_LOG", "info,kube=debug");
+    tracing_subscriber::fmt::init();
+
+    let config = Config::infer().await?;
+
+    // Pick TLS at runtime
+    let use_rustls = std::env::var("USE_RUSTLS").map(|s| s == "1").unwrap_or(false);
+    let client = if use_rustls {
+        let https = {
+            let rustls_config = Arc::new(config.rustls_tls_client_config()?);
+            let mut http = HttpConnector::new();
+            http.enforce_http(false);
+            hyper_rustls::HttpsConnector::from((http, rustls_config))
+        };
+        Client::new(
+            ServiceBuilder::new()
+                .layer(SetBaseUriLayer::new(config.cluster_url))
+                .service(hyper::Client::builder().build(https)),
+        )
+    } else {
+        let https = {
+            let tls = tokio_native_tls::TlsConnector::from(config.native_tls_connector()?);
+            let mut http = HttpConnector::new();
+            http.enforce_http(false);
```

Comment:
This seems like a strange option in `HttpConnector` to have default-true, when you can turn it into an `HttpsConnector` :thinking:

Reply:
Yeah, it looks awkward. But `HttpConnector` is independent and enforcing `http` protocol by default prevents misuse with other protocols like `file://`. I just wish there was a constructor or a builder.